lib/src/version.rs

```diff
@@ -1,18 +1,22 @@
 use crate::errors::{Error, Result};
 use bytes::{BufMut, BytesMut};
 use std::cmp::PartialEq;
-use std::fmt::Debug;
+use std::fmt::{Debug, Display, Formatter};
 
 #[derive(Debug, PartialEq, Eq, Clone, Copy, PartialOrd, Ord)]
 #[non_exhaustive]
 pub enum Version {
     V4,
     V4_1,
+    V4_3,
+    V4_4,
 }
 
 impl Version {
     pub fn add_supported_versions(bytes: &mut BytesMut) {
         bytes.reserve(16);
+        bytes.put_u32(0x0404); // V4_4
+        bytes.put_u32(0x0304); // V4_3
         bytes.put_u32(0x0104); // V4_1
         bytes.put_u32(0x0004); // V4
         bytes.put_u32(0);
```

Comment:
you need to remove the two `put_u32(0)` calls. The initial message only expected 4 versions. I guess on 5.x, the server drains every additional bytes until the next message, on 4.4 those 0 values are part of the next message, effectively sending a few empty chunks.